proposals/vprim/README.md

```diff
@@ -0,0 +1,300 @@
+}
+```
+This adds a nondiegetic overlay and outline for visualizations to a
+grey sphere. The viewer will see the red transparently overlayed, but
+the sphere is still a grey sphere from the perspective of light
+transport (reflected and refracted light).
+
+Consideration must be also given to if or how these schemas affect
+interactive picking. A non-slicing visualization effect might include
+sillhoutte outlines or overlays applied to a `Gprim` to style or
+highlight its importance.
+
+### Example: `VizScreenAnchor`
+For `Gprim`s that need to be aligned, anchored, or otherwise
+nondiegtically constrained for visualization, we suggest that a custom
+`Imageable` be introduced. Just as `UsdGeomPointInstancer` is allowed
+to introduce a break in a transformation hierarchy to introduce
+vectorized instancing, these grouping adapters would be responsible
+for introducing nondiegetic transformations of the existing `Gprim`
+domain (including screen relative placement). We'd propose a Macbeth
+color chart as an example of geometry that users might want
+nondiegetically constrained to a screen position in an asset's OpenUSD.
+```
+(
+    defaultPrim = "Asset"
+)
+
+def Xform "Asset" {
+    def Scope "Geometry" {
+        def Sphere "sphere" {}
+    }
+}
+
+def VizAnchor "Visualization" {
+    # Collection of properties that anchor descendants to 
+    # the lower right corner of screen
+    def Plane "MacbethChart" {}
+}
+```
+
+A nondiegetic transformation could be used to orient an object to
+ camera or provide a floating region for a label around an object.
+
+### Sample Set of Visualization Primitives
+A proposed initial set of visualization primitives is listed below.
+Additional iterations of this document should ensure that the the
+visualization primitive concept can accomodate existing proposals.
+Further evaluation should be done to see if existing `Gprim`s might be
+more appropriate as `Vprim`s (like `HermiteCurves` which is explicitly
+called out as "not for final rendering").
+* `VizPoints`: Points with 3-D positions but screen space widths
+* `VizVolumeSamples`: List of 3-D points at which to sample fields in
+a `UsdVol` volume. The field type can drive the output style (`vector`
+renders as arrows, `density` or `color` render as screen space points)
+or could be otherwise configured.
+* `VizPolylines`: Align with existing curve styling proposals
+* `VizPaths`: Cubically interpolated curves that align with existing
+curve styling proposals
+* `VizLabel`: Align with existing text schema proposals
+* `VizOverlayAPI`: Add a nondiegetic color overlay on top of an
+existing `Gprim`
+* `VizSilhoutteAPI`: Add a nondiegetic edge on top of an existing
+`Gprim`
+* `VizScreenAnchor`: Nondiegetic screen anchoring transformation
+
+### Schema Slicing
+One important consideration when introducing schemas in this domain is
+to whether or not they are new drawables (`Vprim`s) or styles on top
+of existing drawables (`Gprim` + `VizStyleAPI`).
```

Comment:
I understand the concerns about schema slicing, but I also think that this approach will require (in the long term) a ton of duplicate schema definitions for everything in the Gprim hierarchy, which I don't love.  At the hydra level, I'd plan to implement these with the same usdImaging and hydra code and set appropriate flags for xform computation and lighting participation based on Vprim/Gprim; I'd strongly resist much duplication there because the maintenance burden would be significant.

Reply:
The intent is definitely not to duplicate the `Gprim` hierarchy.  To me, the main grey area (with the current hierarchy) is how to handle screen dependent points and curves.  With curves, there are already multiple schema definitions and they're already partitioned between those intended for production rendering (`BasisCurves`) and annotation and rigging purposes (`HermiteCurves` and `NurbsCurves`).

However, that doesn't necessarily mean that `Vprim` is necessarily the right answer to that problem.  Without introducing a higher level `Vprim` class, you could disambiguate curves that are "wires" and "tubes/ribbons".  Wires without widths would be "intrinsically boundable" by their points and a "styling API" could be used to give wires nondiegetic effects like pixel widths.  This would (perhaps narrowly) avoid a "slice" (like interpreting a single pixel width as a meter width) because consumers unaware of or unable to process styles would simply process the curve as an infinitely thin wire.  I'd argue that the disambiguation between "wires" and "non-wires" would be best done at the schema level, but presence of `widths` and `normals` could be used as well.  (I suspect that if "wire-ness" is not specified at the schema level, validators will flag curves authored without `widths` as errors which is why I'd lean that way.)